core/controllers/admin_test.py

```diff
@@ -1739,3 +1739,50 @@ def test_check_community_reviewer_rights_without_username(self):
 
         self.assertEqual(response['error'], 'Missing username param')
         self.logout()
+
+
+class IntereactionByExplorationIdHandlerTests(test_utils.GenericTestBase):
+    """Tests for interaction by exploration handler."""
+
+    EXP_ID = 'exp'
+
+    def setUp(self):
+        """Complete the signup process for self.ADMIN_EMAIL."""
+        super(IntereactionByExplorationIdHandlerTests, self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+        self.exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.editor_id, end_state_name='End')
+
+    def test_interaction_by_exploration_id_handler(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+
+        # Test that it returns all interaction ids.
+        payload = {
+            'exploration_id': self.EXP_ID
+        }
+
+        response = self.get_json(
+            '/interactionsbyexplorationid', params=payload)
+        interactions_list = response['interactions']
+        self.assertEqual(interactions_list, ['EndExploration', 'TextInput'])
+
+    def test_interaction_id_handler_with_invalid_id(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+
+        payload = {
+            'exploration_id': 'invalid_exp'
+        }
+
+        response = self.get_json(
+            '/interactionsbyexplorationid', params=payload,
+            expected_status_int=400)
+        self.assertEqual(response['error'], 'Exploration does not exist.')
+
+    def test_interaction_id_handler_with_no_id(self):
```

Comment:
Add a separate test for non-string exp_id.

Reply:
(u'0', <type 'unicode'>)
I'm getting type as Unicode when using int or any other type @DubeySandeep do i need something else to parse this Unicode ?

Reply:
`elif not isinstance(exploration_id, python_utils.BASESTRING):`
i even tried above thing but can't check for check int

Reply:
@DubeySandeep ??

Reply:
Firstly, (as suggested before) please assign me to the PR if you want m to review/reply on some thread!

Have you tried printing the type of exploration_id?

Reply:
actually I cannot you assign the PR don't know why and yes the type was Unicode and I couldn't find anything unusual in the other handler there was supposed to be an error handler that I did add but fixed error returned to be in JSON but can't find anything other than that.

Reply:
Have you tried python_utils.UNICODE instead of BASESTRING?

Reply:
Now i have used method which you suggested on Pratyush Pr and it works fine i guess although I'll try with Unicode as well

Reply:
@DubeySandeep even if I do the above I don't get whether that type is string or int

Reply:
I can only check that it is not a unicode but that is not sufficient i guess